### DISCUSSION TITLE
How to enable multiple application urls on different port with same scheme?

### DISCUSSION THREAD

**Discussion by javafun** [microsoft/aspire#5338] [General]
Hi

I worked on a project that supports multisite, and I'm currently specifying different port in application url to simulate multi-site testing locally. However, when I started the Aspire host app, it will always use the **last** application url e.g. `https://localhost:7145`. Is there any way I can enable both in Aspire host? (Note: I can see different scheme will be able to show two urls in Aspire dashboard, but it's not flexible enough if I want to test more than 2 sites)

```
    "https": {
      "commandName": "Project",
      "dotnetRunMessages": true,
      "launchBrowser": true,
      "applicationUrl": "https://localhost:7144;https://localhost:7145",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    }
```

**Comment by davidfowl** [comment]
cc @davidebbo can you take a look?

**Comment by davidebbo** [comment]
This is happening because we use the scheme as the endpoint name: https://github.com/dotnet/aspire/blob/319bcbb1eee9f6a6d4a64ddf67aff03bd337bef9/src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs#L409

So when the second https profile endpoint comes, instead of creating a new one, we just update the existing one (that's why we end up with the last one winning). This does seem like a bug, since in non-Aspire app, both endpoints are honored.

I need to think through this some more. I suppose we could keep the first one named https, and come up with name variations for the rest.

But that could have some implications on publishing, especially with ACA only supporting one external endpoint. Or maybe this is a runmode thing only, and we don't mess with publishing? Though that might be quirky.

@davidfowl your thoughts?

**Reply by davidfowl** [reply]
i think we’d need to do just that (create names). Gotta think through the naming scheme

**Reply by davidebbo** [reply]
> i think we’d need to do just that (create names). Gotta think through the naming scheme

For both Run and Publish modes, right?

**Reply by davidfowl** [reply]
Yes in both modes.

**Comment by davidebbo** [comment]
Also, worth noting that Kestrel endpoints don't have this issue, since they are named. So @javafun, you could work around by adding the following to your `appsettings.json`, which will override the profile `applicationUrl`:

```json
  "Kestrel": {
    "Endpoints": {
      "App1": { "Url": "https://*:7144" },
      "App2": { "Url": "https://*:7145" }
    }
  }
```

Can you give this a try?

**Reply by javafun** [reply]
Hi @davidebbo Apologies for delayed response. Yes, this alternative works for me.

**Comment by davidebbo** [comment]
After discussing with @davidfowl on https://discord.com/channels/732297728826277939/1277976825703698494, we decided to only do this for Run mode, because there is no easy way to do it meaningfully in Publish mode.

So at least, this lets us preserver the non-Aspire runtime behavior. But if the user wants to go further in this direction, they should do it in a more complete way, as demonstrated in this sample: https://github.com/dotnet/aspire-samples/tree/main/samples/HealthChecksUI

PR #5481 sent.

**Reply by davidfowl** [reply]
Merged!

**Comment by javafun** [comment]
Has this fix been released? I'm currently using the latest 8.2.0 aspire, but issue seems still exist. 

Thanks

**Reply by davidfowl** [reply]
No it’s part of the 9.0 release. @joperezr are we only back porting regressions to the 8.2 branch?

**Reply by joperezr** [reply]
Yes. 8.2 branch will mostly just get dependency updates and regression fixes. The rest should go to 9.0

**Reply by davidebbo** [reply]
@javafun you may want to try the daily builds to verify the fix: https://github.com/dotnet/aspire/blob/main/docs/using-latest-daily.md
